test/torchaudio_unittest/datasets/utils_test.py

```diff
@@ -1,36 +1,40 @@
+import torch
 from torchaudio_unittest.common_utils import (
     TorchaudioTestCase,
-    get_asset_path,
+    TempDirMixin,
+    get_asset_path
 )
 
 from torchaudio.datasets import utils as dataset_utils
-from torchaudio.datasets.commonvoice import COMMONVOICE
 
-original_ext_audio = COMMONVOICE._ext_audio
 
+class Dataset(torch.utils.data.Dataset):
+    def __getitem__(self, n):
+        return torch.randn(32, 2, 256), 2
 
-class TestIterator(TorchaudioTestCase):
-    @classmethod
-    def setUpClass(cls):
-        COMMONVOICE._ext_audio = ".wav"
+    def __len__(self) -> int:
+        return 32
+
+    def __iter__(self):
+        return iter(range(32))
 
+
+class TestIterator(TorchaudioTestCase, TempDirMixin):
     @classmethod
-    def tearDownClass(cls):
-        COMMONVOICE._ext_audio = original_ext_audio
+    def setUpClass(cls):
+        cls.location = cls.get_base_temp_dir()
+        cls.dataset = Dataset()
```

Comment:
Since there is no requirement to share these attributes across the test cases, they should be defined and used in each test case.

```python
def test_diskcache_iterator(self)
    data = dataset_utils.diskcache_iterator(Dataset(), location=self.get_temp_path())
    ...

def test_bg_iterator(self):
    data = dataset_utils.bg_iterator(Dataset())
    ...
```